Build failure: rusty-man

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

`nix build github:nixos/nixpkgs/nixos-unstable#rusty-man`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318186163

### Relevant log output

```console
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/n6641jmsnxzps606x5d6jxi112fp82bc-source
source root is source
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/Cargo.lock and /build/rusty-man-0.5.0-vendor/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing cargoBuildHook
cargoBuildHook flags: -j 24 --target x86_64-unknown-linux-gnu --offline --profile release
[1m[92m   Compiling[0m proc-macro2 v1.0.27
[1m[92m   Compiling[0m unicode-xid v0.2.2
[1m[92m   Compiling[0m syn v1.0.73
[1m[92m   Compiling[0m libc v0.2.97
[1m[92m   Compiling[0m autocfg v1.0.1
[1m[92m   Compiling[0m getrandom v0.1.16
[1m[92m   Compiling[0m cfg-if v1.0.0
[1m[92m   Compiling[0m ppv-lite86 v0.2.10
[1m[92m   Compiling[0m siphasher v0.3.5
[1m[92m   Compiling[0m serde_derive v1.0.126
[1m[92m   Compiling[0m version_check v0.9.3
[1m[92m   Compiling[0m lazy_static v1.4.0
[1m[92m   Compiling[0m serde v1.0.126
[1m[92m   Compiling[0m log v0.4.14
[1m[92m   Compiling[0m proc-macro-hack v0.5.19
[1m[92m   Compiling[0m ident_case v1.0.1
[1m[92m   Compiling[0m strsim v0.9.3
[1m[92m   Compiling[0m new_debug_unreachable v1.0.4
[1m[92m   Compiling[0m fnv v1.0.7
[1m[92m   Compiling[0m stable_deref_trait v1.2.0
[1m[92m   Compiling[0m bitflags v1.2.1
[1m[92m   Compiling[0m unicode-width v0.1.8
[1m[92m   Compiling[0m mac v0.1.1
[1m[92m   Compiling[0m precomputed-hash v0.1.1
[1m[92m   Compiling[0m utf-8 v0.7.6
[1m[92m   Compiling[0m futf v0.1.4
[1m[92m   Compiling[0m crossbeam-utils v0.8.5
[1m[92m   Compiling[0m matches v0.1.8
[1m[92m   Compiling[0m itoa v0.4.7
[1m[92m   Compiling[0m once_cell v1.8.0
[1m[92m   Compiling[0m pkg-config v0.3.19
[1m[92m   Compiling[0m array-macro v1.0.5
[1m[92m   Compiling[0m cc v1.0.68
[1m[92m   Compiling[0m owning_ref v0.4.1
[1m[92m   Compiling[0m phf_shared v0.8.0
[1m[92m   Compiling[0m xi-unicode v0.3.0
[1m[92m   Compiling[0m tendril v0.4.2
[1m[92m   Compiling[0m unicode-segmentation v1.7.1
[1m[92m   Compiling[0m ryu v1.0.5
[1m[92m   Compiling[0m crc32fast v1.2.1
[1m[92m   Compiling[0m smallvec v1.6.1
[1m[92m   Compiling[0m signal-hook v0.3.9
[1m[92m   Compiling[0m tinyvec_macros v0.1.0
[1m[92m   Compiling[0m dtoa v0.4.8
[1m[92m   Compiling[0m adler v1.0.2
[1m[92m   Compiling[0m hashbrown v0.9.1
[1m[92m   Compiling[0m serde_json v1.0.64
[1m[92m   Compiling[0m safemem v0.3.3
[1m[92m   Compiling[0m num-traits v0.2.14
[1m[92m   Compiling[0m num-integer v0.1.44
[1m[92m   Compiling[0m num-iter v0.1.42
[1m[92m   Compiling[0m num-rational v0.3.2
[1m[92m   Compiling[0m proc-macro-error-attr v1.0.4
[1m[92m   Compiling[0m ahash v0.6.3
[1m[92m   Compiling[0m proc-macro-error v1.0.4
[1m[92m   Compiling[0m indexmap v1.6.2
[1m[92m   Compiling[0m miniz_oxide v0.4.4
[1m[92m   Compiling[0m dtoa-short v0.3.3
[1m[92m   Compiling[0m line-wrap v0.1.1
[1m[92m   Compiling[0m tinyvec v1.2.0
[1m[92m   Compiling[0m base64 v0.13.0
[1m[92m   Compiling[0m xml-rs v0.8.3
[1m[92m   Compiling[0m numtoa v0.1.0
[1m[92m   Compiling[0m same-file v1.0.6
[1m[92m   Compiling[0m crossbeam-channel v0.5.1
[1m[92m   Compiling[0m convert_case v0.4.0
[1m[92m   Compiling[0m byteorder v1.4.3
[1m[92m   Compiling[0m nodrop v0.1.14
[1m[92m   Compiling[0m walkdir v2.3.2
[1m[92m   Compiling[0m unicode-bidi v0.3.5
[1m[92m   Compiling[0m quote v1.0.9
[1m[92m   Compiling[0m servo_arc v0.1.1
[1m[92m   Compiling[0m time v0.1.43
[1m[92m   Compiling[0m getrandom v0.2.3
[1m[92m   Compiling[0m rand_core v0.5.1
[1m[92m   Compiling[0m signal-hook-registry v1.4.0
[1m[92m   Compiling[0m fxhash v0.2.1
[1m[92m   Compiling[0m termion v1.5.6
[1m[92m   Compiling[0m heck v0.3.3
[1m[92m   Compiling[0m textwrap v0.11.0
[1m[92m   Compiling[0m anyhow v1.0.41
[1m[92m   Compiling[0m thin-slice v0.1.1
[1m[92m   Compiling[0m rand_chacha v0.2.2
[1m[92m   Compiling[0m rand_pcg v0.2.1
[1m[92m   Compiling[0m lazycell v1.3.0
[1m[92m   Compiling[0m regex-syntax v0.6.25
[1m[92m   Compiling[0m flate2 v1.0.20
[1m[92m   Compiling[0m rand v0.7.3
[1m[92m   Compiling[0m onig_sys v69.7.0
[1m[92m   Compiling[0m percent-encoding v2.1.0
[1m[92m   Compiling[0m num-complex v0.3.1
[1m[92m   Compiling[0m form_urlencoded v1.0.1
[1m[92m   Compiling[0m clap v2.33.3
[1m[92m   Compiling[0m errno v0.2.7
[1m[92m   Compiling[0m unicode-normalization v0.1.19
[1m[92m   Compiling[0m chrono v0.4.19
[1m[92m   Compiling[0m pager v0.16.0
[1m[92m   Compiling[0m env_logger v0.8.4
[1m[92m   Compiling[0m webbrowser v0.5.5
[1m[92m   Compiling[0m xdg v2.2.0
[1m[92m   Compiling[0m phf_generator v0.8.0
[1m[92m   Compiling[0m phf_codegen v0.8.0
[1m[92m   Compiling[0m string_cache_codegen v0.5.1
[1m[92m   Compiling[0m num v0.3.1
[1m[92m   Compiling[0m selectors v0.22.0
[1m[92m   Compiling[0m idna v0.2.3
[1m[92m   Compiling[0m markup5ever v0.10.1
[1m[92m   Compiling[0m url v2.2.2
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'onig_st_init_strend_table_with_size':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:584:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *, st_str_end_key *)' [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   584 |     str_end_cmp,
[1m[33mwarning[0m: onig_sys@69.7.0:       |     ^~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:584:5: note: (near initialization for 'hashType.compare')
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:546:1: note: 'str_end_cmp' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   546 | str_end_cmp(st_str_end_key* x, st_str_end_key* y)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:585:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *)' [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   585 |     str_end_hash,
[1m[33mwarning[0m: onig_sys@69.7.0:       |     ^~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:585:5: note: (near initialization for 'hashType.hash')
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:567:1: note: 'str_end_hash' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   567 | str_end_hash(st_str_end_key* x)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'onig_st_init_callout_name_table_with_size':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:674:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *, st_callout_name_key *)' [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   674 |     callout_name_table_cmp,
[1m[33mwarning[0m: onig_sys@69.7.0:       |     ^~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:674:5: note: (near initialization for 'hashType.compare')
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:633:1: note: 'callout_name_table_cmp' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   633 | callout_name_table_cmp(st_callout_name_key* x, st_callout_name_key* y)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:675:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *)' [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   675 |     callout_name_table_hash,
[1m[33mwarning[0m: onig_sys@69.7.0:       |     ^~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:675:5: note: (near initialization for 'hashType.hash')
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:656:1: note: 'callout_name_table_hash' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   656 | callout_name_table_hash(st_callout_name_key* x)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'names_clear':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:800:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   800 |     onig_st_foreach(t, i_free_name_entry, 0);
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        ^~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        |
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        int (*)(OnigUChar *, NameEntry *, void *) {aka int (*)(unsigned char *, NameEntry *, void *)}
[1m[33mwarning[0m: onig_sys@69.7.0: In file included from oniguruma/src/regparse.c:37:
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, void *)' {aka 'int (*)(unsigned char *, NameEntry *, void *)'}
[1m[33mwarning[0m: onig_sys@69.7.0:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
[1m[33mwarning[0m: onig_sys@69.7.0:       |                               ^~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:35:18: note: in definition of macro '_'
[1m[33mwarning[0m: onig_sys@69.7.0:    35 | # define _(args) args
[1m[33mwarning[0m: onig_sys@69.7.0:       |                  ^~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:785:1: note: 'i_free_name_entry' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   785 | i_free_name_entry(UChar* key, NameEntry* e, void* arg ARG_UNUSED)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'onig_foreach_name':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:869:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   869 |     onig_st_foreach(t, i_names, (HashDataType )&narg);
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        ^~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        |
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        int (*)(OnigUChar *, NameEntry *, INamesArg *) {aka int (*)(unsigned char *, NameEntry *, INamesArg *)}
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, INamesArg *)' {aka 'int (*)(unsigned char *, NameEntry *, INamesArg *)'}
[1m[33mwarning[0m: onig_sys@69.7.0:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
[1m[33mwarning[0m: onig_sys@69.7.0:       |                               ^~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:35:18: note: in definition of macro '_'
[1m[33mwarning[0m: onig_sys@69.7.0:    35 | # define _(args) args
[1m[33mwarning[0m: onig_sys@69.7.0:       |                  ^~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:842:1: note: 'i_names' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   842 | i_names(UChar* key ARG_UNUSED, NameEntry* e, INamesArg* arg)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'onig_renumber_name_table':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:897:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:   897 |     onig_st_foreach(t, i_renumber_name, (HashDataType )map);
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        ^~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        |
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        int (*)(OnigUChar *, NameEntry *, GroupNumMap *) {aka int (*)(unsigned char *, NameEntry *, GroupNumMap *)}
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, GroupNumMap *)' {aka 'int (*)(unsigned char *, NameEntry *, GroupNumMap *)'}
[1m[33mwarning[0m: onig_sys@69.7.0:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
[1m[33mwarning[0m: onig_sys@69.7.0:       |                               ^~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:35:18: note: in definition of macro '_'
[1m[33mwarning[0m: onig_sys@69.7.0:    35 | # define _(args) args
[1m[33mwarning[0m: onig_sys@69.7.0:       |                  ^~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:875:1: note: 'i_renumber_name' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:   875 | i_renumber_name(UChar* key ARG_UNUSED, NameEntry* e, GroupNumMap* map)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'callout_name_table_clear':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:1382:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:  1382 |     onig_st_foreach(t, i_free_callout_name_entry, 0);
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        |
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        int (*)(st_callout_name_key *, CalloutNameEntry *, void *)
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_callout_name_key *, CalloutNameEntry *, void *)'
[1m[33mwarning[0m: onig_sys@69.7.0:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
[1m[33mwarning[0m: onig_sys@69.7.0:       |                               ^~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:35:18: note: in definition of macro '_'
[1m[33mwarning[0m: onig_sys@69.7.0:    35 | # define _(args) args
[1m[33mwarning[0m: onig_sys@69.7.0:       |                  ^~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:1366:1: note: 'i_free_callout_name_entry' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:  1366 | i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'setup_ext_callout_list_values':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:1880:56: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:  1880 |     onig_st_foreach((CalloutTagTable *)ext->tag_table, i_callout_callout_list_set,
[1m[33mwarning[0m: onig_sys@69.7.0:       |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0:       |                                                        |
[1m[33mwarning[0m: onig_sys@69.7.0:       |                                                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
[1m[33mwarning[0m: onig_sys@69.7.0:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
[1m[33mwarning[0m: onig_sys@69.7.0:       |                               ^~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:35:18: note: in definition of macro '_'
[1m[33mwarning[0m: onig_sys@69.7.0:    35 | # define _(args) args
[1m[33mwarning[0m: onig_sys@69.7.0:       |                  ^~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:1862:1: note: 'i_callout_callout_list_set' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:  1862 | i_callout_callout_list_set(UChar* key, CalloutTagVal e, void* arg)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c: In function 'callout_tag_table_clear':
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:1928:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
[1m[33mwarning[0m: onig_sys@69.7.0:  1928 |     onig_st_foreach(t, i_free_callout_tag_entry, 0);
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        ^~~~~~~~~~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        |
[1m[33mwarning[0m: onig_sys@69.7.0:       |                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
[1m[33mwarning[0m: onig_sys@69.7.0:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
[1m[33mwarning[0m: onig_sys@69.7.0:       |                               ^~~~~~~~~~~~~~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/st.h:35:18: note: in definition of macro '_'
[1m[33mwarning[0m: onig_sys@69.7.0:    35 | # define _(args) args
[1m[33mwarning[0m: onig_sys@69.7.0:       |                  ^~~~
[1m[33mwarning[0m: onig_sys@69.7.0: oniguruma/src/regparse.c:1918:1: note: 'i_free_callout_tag_entry' declared here
[1m[33mwarning[0m: onig_sys@69.7.0:  1918 | i_free_callout_tag_entry(UChar* key, CalloutTagVal e, void* arg ARG_UNUSED)
[1m[33mwarning[0m: onig_sys@69.7.0:       | ^~~~~~~~~~~~~~~~~~~~~~~~
[1m[91merror[0m: failed to run custom build command for `onig_sys v69.7.0`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/onig_sys-76cde41c560cbd35/build-script-build` (exit status: 1)
  --- stdout
  TARGET = Some("x86_64-unknown-linux-gnu")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_x86_64-unknown-linux-gnu = None
  CC_x86_64_unknown_linux_gnu = None
  HOST_CC = Some("/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc")
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out/oniguruma/src/regexec.o" "-c" "oniguruma/src/regexec.c"
  exit status: 0
  running: "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out/oniguruma/src/regerror.o" "-c" "oniguruma/src/regerror.c"
  exit status: 0
  running: "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out/oniguruma/src/regparse.o" "-c" "oniguruma/src/regparse.c"
  cargo:warning=oniguruma/src/regparse.c: In function 'onig_st_init_strend_table_with_size':
  cargo:warning=oniguruma/src/regparse.c:584:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *, st_str_end_key *)' [-Wincompatible-pointer-types]
  cargo:warning=  584 |     str_end_cmp,
  cargo:warning=      |     ^~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c:584:5: note: (near initialization for 'hashType.compare')
  cargo:warning=oniguruma/src/regparse.c:546:1: note: 'str_end_cmp' declared here
  cargo:warning=  546 | str_end_cmp(st_str_end_key* x, st_str_end_key* y)
  cargo:warning=      | ^~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c:585:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *)' [-Wincompatible-pointer-types]
  cargo:warning=  585 |     str_end_hash,
  cargo:warning=      |     ^~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c:585:5: note: (near initialization for 'hashType.hash')
  cargo:warning=oniguruma/src/regparse.c:567:1: note: 'str_end_hash' declared here
  cargo:warning=  567 | str_end_hash(st_str_end_key* x)
  cargo:warning=      | ^~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'onig_st_init_callout_name_table_with_size':
  cargo:warning=oniguruma/src/regparse.c:674:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *, st_callout_name_key *)' [-Wincompatible-pointer-types]
  cargo:warning=  674 |     callout_name_table_cmp,
  cargo:warning=      |     ^~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c:674:5: note: (near initialization for 'hashType.compare')
  cargo:warning=oniguruma/src/regparse.c:633:1: note: 'callout_name_table_cmp' declared here
  cargo:warning=  633 | callout_name_table_cmp(st_callout_name_key* x, st_callout_name_key* y)
  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c:675:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *)' [-Wincompatible-pointer-types]
  cargo:warning=  675 |     callout_name_table_hash,
  cargo:warning=      |     ^~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c:675:5: note: (near initialization for 'hashType.hash')
  cargo:warning=oniguruma/src/regparse.c:656:1: note: 'callout_name_table_hash' declared here
  cargo:warning=  656 | callout_name_table_hash(st_callout_name_key* x)
  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'names_clear':
  cargo:warning=oniguruma/src/regparse.c:800:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
  cargo:warning=  800 |     onig_st_foreach(t, i_free_name_entry, 0);
  cargo:warning=      |                        ^~~~~~~~~~~~~~~~~
  cargo:warning=      |                        |
  cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, void *) {aka int (*)(unsigned char *, NameEntry *, void *)}
  cargo:warning=In file included from oniguruma/src/regparse.c:37:
  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, void *)' {aka 'int (*)(unsigned char *, NameEntry *, void *)'}
  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
  cargo:warning=      |                               ^~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
  cargo:warning=   35 | # define _(args) args
  cargo:warning=      |                  ^~~~
  cargo:warning=oniguruma/src/regparse.c:785:1: note: 'i_free_name_entry' declared here
  cargo:warning=  785 | i_free_name_entry(UChar* key, NameEntry* e, void* arg ARG_UNUSED)
  cargo:warning=      | ^~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'onig_foreach_name':
  cargo:warning=oniguruma/src/regparse.c:869:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
  cargo:warning=  869 |     onig_st_foreach(t, i_names, (HashDataType )&narg);
  cargo:warning=      |                        ^~~~~~~
  cargo:warning=      |                        |
  cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, INamesArg *) {aka int (*)(unsigned char *, NameEntry *, INamesArg *)}
  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, INamesArg *)' {aka 'int (*)(unsigned char *, NameEntry *, INamesArg *)'}
  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
  cargo:warning=      |                               ^~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
  cargo:warning=   35 | # define _(args) args
  cargo:warning=      |                  ^~~~
  cargo:warning=oniguruma/src/regparse.c:842:1: note: 'i_names' declared here
  cargo:warning=  842 | i_names(UChar* key ARG_UNUSED, NameEntry* e, INamesArg* arg)
  cargo:warning=      | ^~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'onig_renumber_name_table':
  cargo:warning=oniguruma/src/regparse.c:897:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
  cargo:warning=  897 |     onig_st_foreach(t, i_renumber_name, (HashDataType )map);
  cargo:warning=      |                        ^~~~~~~~~~~~~~~
  cargo:warning=      |                        |
  cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, GroupNumMap *) {aka int (*)(unsigned char *, NameEntry *, GroupNumMap *)}
  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, GroupNumMap *)' {aka 'int (*)(unsigned char *, NameEntry *, GroupNumMap *)'}
  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
  cargo:warning=      |                               ^~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
  cargo:warning=   35 | # define _(args) args
  cargo:warning=      |                  ^~~~
  cargo:warning=oniguruma/src/regparse.c:875:1: note: 'i_renumber_name' declared here
  cargo:warning=  875 | i_renumber_name(UChar* key ARG_UNUSED, NameEntry* e, GroupNumMap* map)
  cargo:warning=      | ^~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'callout_name_table_clear':
  cargo:warning=oniguruma/src/regparse.c:1382:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
  cargo:warning= 1382 |     onig_st_foreach(t, i_free_callout_name_entry, 0);
  cargo:warning=      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=      |                        |
  cargo:warning=      |                        int (*)(st_callout_name_key *, CalloutNameEntry *, void *)
  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_callout_name_key *, CalloutNameEntry *, void *)'
  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
  cargo:warning=      |                               ^~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
  cargo:warning=   35 | # define _(args) args
  cargo:warning=      |                  ^~~~
  cargo:warning=oniguruma/src/regparse.c:1366:1: note: 'i_free_callout_name_entry' declared here
  cargo:warning= 1366 | i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,
  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'setup_ext_callout_list_values':
  cargo:warning=oniguruma/src/regparse.c:1880:56: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
  cargo:warning= 1880 |     onig_st_foreach((CalloutTagTable *)ext->tag_table, i_callout_callout_list_set,
  cargo:warning=      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=      |                                                        |
  cargo:warning=      |                                                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
  cargo:warning=      |                               ^~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
  cargo:warning=   35 | # define _(args) args
  cargo:warning=      |                  ^~~~
  cargo:warning=oniguruma/src/regparse.c:1862:1: note: 'i_callout_callout_list_set' declared here
  cargo:warning= 1862 | i_callout_callout_list_set(UChar* key, CalloutTagVal e, void* arg)
  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/regparse.c: In function 'callout_tag_table_clear':
  cargo:warning=oniguruma/src/regparse.c:1928:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
  cargo:warning= 1928 |     onig_st_foreach(t, i_free_callout_tag_entry, 0);
  cargo:warning=      |                        ^~~~~~~~~~~~~~~~~~~~~~~~
  cargo:warning=      |                        |
  cargo:warning=      |                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
  cargo:warning=      |                               ^~~~~~~~~~~~~~~~
  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'
  cargo:warning=   35 | # define _(args) args
  cargo:warning=      |                  ^~~~
  cargo:warning=oniguruma/src/regparse.c:1918:1: note: 'i_free_callout_tag_entry' declared here
  cargo:warning= 1918 | i_free_callout_tag_entry(UChar* key, CalloutTagVal e, void* arg ARG_UNUSED)
  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~
  exit status: 1

  --- stderr


  error occurred: Command "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-O3" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-6ecaca47b29f290b/out/oniguruma/src/regparse.o" "-c" "oniguruma/src/regparse.c" with args "cc" did not execute successfully (status code exit status: 1).


[1m[33mwarning[0m: build failed, waiting for other jobs to finish...
```

### Additional context

likely related to #475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.1-zen1, NixOS, 26.05 (Yarara), 26.05.20251225.3e2499d`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/24b29jfv23y9mh5ib7a5yljcgm0n65kw-source`

### Notify maintainers

@Defelo 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ru/rusty-man/package.nix
+++ b/pkgs/by-name/ru/rusty-man/package.nix
@@ -19,6 +19,9 @@ rustPlatform.buildRustPackage (finalAttrs: {
 
   cargoHash = "sha256-ZIRwp5AJugMDxg3DyFIH5VlD0m4Si2tJdspKE5QEB4M=";
 
+  # Fix build with gcc 15
+  env.NIX_CFLAGS_COMPILE = "-std=gnu17";
+
   nativeInstallCheckInputs = [ versionCheckHook ];
   doInstallCheck = true;
 
@@ -30,8 +33,6 @@ rustPlatform.buildRustPackage (finalAttrs: {
     homepage = "https://git.sr.ht/~ireas/rusty-man";
     changelog = "https://git.sr.ht/~ireas/rusty-man/tree/v${finalAttrs.version}/item/CHANGELOG.md";
     license = lib.licenses.mit;
-    maintainers = with lib.maintainers; [
-      defelo
-    ];
+    maintainers = with lib.maintainers; [ defelo ];
   };
 })

